### PR TITLE
Upgrade from Ignition v1 to 2.0.0 and Fuze

### PR DESCRIPTION
**Pull request by dghubble**
* By default, templates for Ignition are rendered and parsed as github.com/coreos/fuze YAML configs, which formalize the transform from YAML to Ignition JSON
* Ignition files (.ign/.ignition) should be validated and served directly, without template rendering
* Remove support for Ignition v1
* This change is breaking! Users need to tweak any YAML Ignition templates to the very similar Fuze YAML format
* With this change, reference clusters will be provisioned by Ignition 2.0.0. Ignition v1 is no longer supported.

rel: #141 #188

### CODE REVIEW

**Comment by crawford** [comment]
Wow, this is so much simpler.

File: bootcfg/http/ignition.go [hunk 1]
```diff
@@ -39,6 +37,20 @@ func ignitionHandler(srv server.Server) ContextHandler {
 			return
 		}
 
+		// Skip rendering if raw Ignition JSON is provided
+		if isIgnition(profile.IgnitionId) {
+			cfg, err := ignition.Parse([]byte(contents))
```
Comment (dghubble) [inline bootcfg/http/ignition.go]: The (rare) user who decides to supply raw Ignition JSON must write an Ignition config whose version is equal to or less than the Ignition.Parse vendored with that release of bootcfg.

Comment (dghubble) [inline bootcfg/http/ignition.go]: After discussion, we'll pass through the contents in all cases and just log any parse errors as warnings.

**Comment by dghubble** [comment]
Besides changelog/doc updates, I believe this is ok, when our dependents are ready.

File: Documentation/ignition.md [hunk 2]
```diff
@@ -1,42 +1,82 @@
...
-            DNS={{.networkd_dns}}
-            Address={{.networkd_address}}
-    {{ if .ssh_authorized_keys }}
+    storage:
+      disks:
+        - device: /dev/sda
+          wipe_table: true
+          partitions:
+            - label: ROOT
+              number: 0
```
Comment (crawford) [inline Documentation/ignition.md]: It should be safe to omit this line.

File: Documentation/ignition.md [hunk 3]
```diff
@@ -82,7 +165,20 @@ ignition/run-hello.ign:
...
+In v0.4.0, `bootcfg` switched to using the CoreOS [fuze](https://github.com/coreos/fuze) library, which formalizes and improves upong the YAML to Ignition JSON transform. Fuze provides better support for Ignition 2.0.0+, handles file content encoding, patches Ignition bugs, and performs better validations.
+
+To upgrade to bootcfg `v0.4.0`, upgrade your Ignition YAML templates to match the [Fuze config schema](https://github.com/coreos/fuze/blob/master/doc/configuration.md). CoreOS machines **must be 1010.1.0 or newer**, support for the older Ignition v1 format has been dropped.
+
+Typically, you'll need to do the following:
+
+* Remove `ignition_version: 1`, Fuze configs are versionless.
+* Update `filesystems` section and set the `name`
+* Update `files` section to use `inline` as shown above.
+* Replace `uid` and `gid` with ``
```
Comment (crawford) [inline Documentation/ignition.md]: With what?

Comment (dghubble) [inline Documentation/ignition.md]: Fixed, thanks.

File: Documentation/ignition.md (hunk 2 above)
Comment (dghubble) [inline Documentation/ignition.md]: Removed.

**Pull request merged**